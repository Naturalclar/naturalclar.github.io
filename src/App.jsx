import React from 'react';
import Radium, { StyleRoot } from 'radium';
import { BrowserRouter as Router, Route, NavLink, Switch } from 'react-router-dom';
import Header from './components/Header';
import Footer from './components/Footer';
import Menu from './components/Menu';
import MenuItem from './components/MenuItem';
import Color from './utils/Color';
import Home from './components/pages/Home';
import About from './components/pages/About';
import Contact from './components/pages/Contact';

const styles = {
  body: {
    color: Color.TEXT_PRIMARY,
    display: 'grid',
    gridTemplateRows: 'auto auto 1fr auto',
    gridTemplateAreas: '"header" "menu" "content" "footer"',
    height: '100%',
    width: '100%',
    position: 'absolute',
  },
  header: {
    gridArea: 'header',
  },
  footer: {
    gridArea: 'footer',
  },
  menu: {
    gridArea: 'menu',
  },
  content: {
    gridArea: 'content',
  },
  link: {
    color: 'inherit',
    textDecoration: 'none',
  },
  active: {
    textShadow: `0 0 1rem ${Color.TEXT_SHADOW},0 0 0.2rem ${Color.TEXT_SHADOW}`,
  },
};

function App() {
  const menu = [
    {
      title: 'Home',
      path: '/',
    },
    {
      title: 'About me',
      path: '/about',
    },
    {
      title: 'Contact me',
      path: '/contact',
    },
  ];
  const menuItems = menu.map(val => (
    <MenuItem key={val.title}>
      <NavLink
        style={styles.link}
        activeStyle={styles.active}
        key={val.title}
        to={val.path}
        href={val.path}
        exact
      >{val.title}
      </NavLink>
    </MenuItem>
  ));
  return (
    <StyleRoot>
      <Router>
        <div style={styles.body}>
          <Header title="naturalclar" style={styles.header} />
          <Menu style={styles.menu}>
            {menuItems}
          </Menu>
          <Switch>
            <Route path="/" exact component={Home} />
            <Route path="/about" component={About} />
            <Route path="/contact" component={Contact} />
          </Switch>
          <Footer
            title="naturalclar"
            style={styles.footer}
            color={Color.TEXT_PRIMARY}
            name="Jesse K."
          />
        </div>
      </Router>
    </StyleRoot>
  );
}


export default Radium(App);
