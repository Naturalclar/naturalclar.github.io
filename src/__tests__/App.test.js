import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

const wrapper = shallow(<App />)

it('renders without crashing', () => {
  expect(wrapper.length).toBe(1)
})

it('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
