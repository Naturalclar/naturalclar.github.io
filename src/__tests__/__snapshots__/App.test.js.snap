// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: FontAwesome;
  padding: 0.3rem;
  margin: 0.3rem;
  color: rgba(255,255,255,0.8);
  background: rgba(0,0,0,0.1);
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 50%;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c10:hover {
  box-shadow: 2px 2px 2px rgba(0,0,0,0.3);
  background: rgba(0,0,0,0.2);
  color: white;
}

.c7 {
  grid-area: footer;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 2rem;
  color: white;
}

.c7 p {
  padding: 1rem 0;
}

.c8 {
  width: 90%;
  height: 1px;
  background: white;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "Indie Flower",cursive;
  color: inherit;
  font-size: 2rem;
  padding: 1rem 0;
}

.c2 {
  text-align: center;
  list-style-type: none;
  grid-area: menu;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin: 1rem;
  cursor: pointer;
  font-size: 1.5rem;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c3:hover {
  -webkit-transform: scale(1.3);
  -ms-transform: scale(1.3);
  transform: scale(1.3);
  text-shadow: 0 0 1rem dodgerblue,0 0 0.2rem dodgerblue;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  -webkit-transition: all 0.3 ease-in-out;
  transition: all 0.3 ease-in-out;
  font-family: 'Indie Flower',cursive;
  opacity: 0;
  -webkit-animation: kVINEQ 0.4s ease both;
  animation: kVINEQ 0.4s ease both;
}

.c6 h1 {
  margin: 2rem 0 font-size:2.5rem;
}

.c6 h2 {
  font-size: 1.5rem;
}

.c0 {
  color: mintcream;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  position: absolute;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Indie Flower',none;
}

.c4.active {
  text-shadow: 0 0 1rem dodgerblue,0 0 0.2rem dodgerblue;
}

<div
  data-radium={true}
>
  <div
    className="c0"
  >
    <header
      className="c1"
    >
      naturalclar
    </header>
    <ul
      className="c2"
    >
      <li
        className="c3"
      >
        <a
          aria-current="page"
          className="c4 active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          aria-current={null}
          className="c4"
          href="/about"
          onClick={[Function]}
        >
          About me
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          aria-current={null}
          className="c4"
          href="/contact"
          onClick={[Function]}
        >
          Contact me
        </a>
      </li>
    </ul>
    <main
      className="c5"
    >
      <div
        className="c6"
      >
        <h1>
          Jesse K.
        </h1>
        <h2>
          Front-End Developer
        </h2>
      </div>
    </main>
    <footer
      className="c7"
    >
      <div
        className="c8"
      />
      <ul
        className="c9"
      >
        <div
          data-radium={true}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <a
            className="c10"
            href="mailto:jesse.katsumata@gmail.com"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </div>
        <div
          data-radium={true}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <a
            className="c10"
            href="https://www.linkedin.com/in/jesse-katsumata-1a686462/"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </div>
        <div
          data-radium={true}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <a
            className="c10"
            href="https://twitter.com/natural_clar"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </div>
        <div
          data-radium={true}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <a
            className="c10"
            href="https://github.com/Naturalclar"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </div>
        <div
          data-radium={true}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <a
            className="c10"
            href="https://codepen.io/naturalclar"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </div>
      </ul>
      <p>
        naturalclar
         - Created by 
        Jesse K.
         © 2017
      </p>
    </footer>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
